Failed to build the connector with OpenSSL on M1

```
$ brew install openssl@1.1
$ make test
Running all packages tests
go clean -testcache
go test -tags "" ./... -v -p 1
go: downloading github.com/stretchr/testify v1.7.1
go: downloading github.com/google/uuid v1.3.0
go: downloading github.com/shopspring/decimal v1.3.1
go: downloading gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c
go: downloading github.com/davecgh/go-spew v1.1.0
# github.com/tarantool/go-tarantool.test
/opt/homebrew/Cellar/go/1.19.3/libexec/pkg/tool/darwin_arm64/link: running clang failed: exit status 1
ld: warning: ignoring file /usr/local/opt/openssl@1.1/lib/libssl.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
ld: warning: ignoring file /usr/local/opt/openssl@1.1/lib/libcrypto.dylib, building for macOS-arm64 but attempting to link with file built for macOS-x86_64
Undefined symbols for architecture arm64:
  "_ASN1_INTEGER_free", referenced from:
      __cgo_3cad583f5b97_Cfunc_ASN1_INTEGER_free in 000009.o
     (maybe you meant: __cgo_3cad583f5b97_Cfunc_ASN1_INTEGER_free)
  "_ASN1_INTEGER_new", referenced from:
      __cgo_3cad583f5b97_Cfunc_ASN1_INTEGER_new in 000009.o
     (maybe you meant: __cgo_3cad583f5b97_Cfunc_ASN1_INTEGER_new)
  "_ASN1_INTEGER_to_BN", referenced from:
      __cgo_3cad583f5b97_Cfunc_ASN1_INTEGER_to_BN in 000009.o
     (maybe you meant: __cgo_3cad583f5b97_Cfunc_ASN1_INTEGER_to_BN)
  "_ASN1_OCTET_STRING_new", referenced from:
      _add_custom_ext in 000032.o
  "_ASN1_OCTET_STRING_set", referenced from:
      _add_custom_ext in 000032.o
  "_BIO_clear_flags", referenced from:
      _x_bio_create in 000034.o
      _X_BIO_clear_flags in 000034.o
     (maybe you meant: _X_BIO_clear_flags, __cgo_3cad583f5b97_Cfunc_X_BIO_clear_flags )
```

It is possible to use the connector with disabled SSL support with `go_tarantool_ssl_disable` build tag:
```
$ make TAGS="go_tarantool_ssl_disable" test
$ go test -v -tags go_tarantool_ssl_disable ./... -v -p 1
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -39,6 +39,7 @@ The release improves compatibility with new Tarantool versions.
   package-level variable decimalPrecision (#233)
 - Flaky test TestClientRequestObjectsWithContext (#244)
 - Flaky test multi/TestDisconnectAll (#234)
+- Build on macOS with Apple M1 (#260)
 
 ## [1.9.0] - 2022-11-02
 

--- a/go.mod
+++ b/go.mod
@@ -9,7 +9,7 @@ require (
 	github.com/pkg/errors v0.9.1 // indirect
 	github.com/shopspring/decimal v1.3.1
 	github.com/stretchr/testify v1.7.1 // indirect
-	github.com/tarantool/go-openssl v0.0.8-0.20220711094538-d93c1eff4f49
+	github.com/tarantool/go-openssl v0.0.8-0.20230307065445-720eeb389195
 	google.golang.org/appengine v1.6.7 // indirect
 	gopkg.in/check.v1 v1.0.0-20200227125254-8fa46927fb4f // indirect
 	gopkg.in/vmihailenco/msgpack.v2 v2.9.2

